src/task/executable_task.rs

```diff
@@ -384,18 +393,72 @@ fn get_output_writer_and_handle() -> (ShellPipeWriter, JoinHandle<String>) {
     (writer, handle)
 }
 
-/// Task specific environment variables.
-fn get_export_specific_task_env(task: &Task) -> String {
-    // Append the environment variables if they don't exist
+/// Get the environment variable based on their priority
+fn get_export_specific_task_env(task: &Task, command_env: IndexMap<String, String>) -> String {
+    // Early return if task.env() is empty
+    if task.env().is_none_or(|map| map.is_empty()) {
+        return String::new();
+    }
+
     let mut export = String::new();
+    let mut export_merged: HashMap<String, String> = HashMap::new();
+
+    // Define keys that should not be overridden
+    let override_excluded_keys: HashSet<&str> = [
+        "PIXI_PROJECT_ROOT",
+        "PIXI_PROJECT_NAME",
+        "PIXI_PROJECT_VERSION",
+        "PIXI_PROMPT",
+        "PIXI_ENVIRONMENT_NAME",
+        "PIXI_ENVIRONMENT_PLATFORMS",
+        "CONDA_PREFIX",
+        "CONDA_DEFAULT_ENV",
+        "PATH",
+        "INIT_CWD",
+        "PWD",
+    ]
+    .iter()
+    .cloned()
+    .collect();
+
+    // Create environment map for priority-based merging.
+    // Considering future extensions, there may be other types of environment variables.
+    let mut env_map: HashMap<&'static str, Option<IndexMap<String, String>>> = HashMap::new();
+    // Command env variables
+    env_map.insert("COMMAND_ENV", Some(command_env.clone()));
+    // Task specific environment variables
+    env_map.insert(
+        "TASK_SPECIFIC_ENVS",
```

Comment:
`env_map` will always contain the same two keys right?

Then, you should just use a struct similar to this.
```rust
struct EnvMap {
    command_env: IndexMap<String, String>
    task_specific_env: IndexMap<String, String>
}
```

You should also be able to put some logic like the priority as methods of `EnvMap`

Reply:
Or do you need a map/struct at all?
Can't they just be two local variables?

Reply:
Ahhh, my data structure choice is based on extensible need, as there may be other kinds of environment variables in the future. I don't need to manually "A.extend(B).extend(C).extend(D)".